lambda_src/drivers/process_https.py

```diff
@@ -59,22 +58,28 @@ def process_row(
     req_headers.setdefault('User-Agent', 'GEFF 1.0')
     req_headers.setdefault('Accept-Encoding', 'gzip')
 
+    # We look for an auth header and if found, we parse it from its encoded format
     if auth is not None:
         auth = decrypt_if_encrypted(auth)
-        assert auth is not None
+
         req_auth = (
             loads(auth)
-            if auth.startswith('{')
+            if auth and auth.startswith('{')
```

Comment:
if the assert helps the type system know this is Ok without explicitly stating it that seems better -- maybe add a comment that this is why it's there?

Reply:
Sorry I don't think I fully understood the requested change here. Just to confirm, you're asking to bring back the assert and remove the `None` check and add a comment as to why the assert is there? 

assert fail causes an error and `if auth` check only checks for a curly branch when auth is not `None`. Isn't that the behavior we want here?

Reply:
yeah -- I think the extra check is vestigial since we know it's not going to `String` and the `assert` was there to make sure the pypi type system doesn't complain since it doesn't have a way of indicating that the decrypt will return None only if it's passed None.